include/logging/log_core.h

```diff
@@ -232,12 +231,15 @@ extern "C" {
 /******************************************************************************/
 /****************** Defiinitions used by minimal logging **********************/
 /******************************************************************************/
-void log_minimal_hexdump_print(int level, const void *data, size_t size);
-
-#define Z_LOG_TO_PRINTK(_level, fmt, ...) do {				     \
-		printk("%c: " fmt "\n", z_log_minimal_level_to_char(_level), \
-			##__VA_ARGS__);					     \
-	} while (false)
+void z_log_minimal_hexdump_print(int level, const void *data, size_t size);
```

Comment:
shouldn't these declarations be under an `ifdef CONFIG_LOG_MINIMAL` ? These functions are implemented only under this condition.

Reply:
i don't know where it is written but i think we shouldn't ifdef api only the implementation which allows to use `IS_ENABLED()` and still have api visible.